### ISSUE TITLE
Heading values that are passed to Valhalla are not being respected.

### ISSUE THREAD

**Issue by Jkkarr2015** [valhalla/valhalla#1980]
I was doing some testing yesterday and found that the heading that I pass in is not being used it seems. I used this query: 

valhalla_run_route -j '{"locations":[{"lat":38.3100868, "lon":-85.8449319, "heading":319.01294}, {"lat":38.330871, "lon":-85.849811}], "costing":"auto", "directions_options":{"units":"miles"}}' --config valhalla.json

For reference, I was heading Northwest on State St. and had just past West Daisy Lane which was on my right. I got these instructions.

2019/10/18 12:50:48.753462 [NARRATIVE] From: ,,,,,
2019/10/18 12:50:48.753473 [NARRATIVE] To: ,,,,,
2019/10/18 12:50:48.753485 [NARRATIVE] ==============================================
2019/10/18 12:50:48.753506 [NARRATIVE] 1: Drive southeast on State Street. | 0.1 mi
2019/10/18 12:50:48.753518 [NARRATIVE]    VERBAL_PRE: Drive southeast on State Street for 500 feet. Then Turn left onto West Daisy Lane.
2019/10/18 12:50:48.753527 [NARRATIVE] ----------------------------------------------
2019/10/18 12:50:48.753539 [NARRATIVE] 2: Turn left onto West Daisy Lane. | 0.4 mi
2019/10/18 12:50:48.753550 [NARRATIVE]    VERBAL_ALERT: Turn left onto West Daisy Lane.
2019/10/18 12:50:48.753559 [NARRATIVE]    VERBAL_PRE: Turn left onto West Daisy Lane.
2019/10/18 12:50:48.753569 [NARRATIVE]    VERBAL_POST: Continue for 4 tenths of a mile.
2019/10/18 12:50:48.753577 [NARRATIVE] ----------------------------------------------
2019/10/18 12:50:48.753591 [NARRATIVE] 3: Turn left onto Green Valley Road. | 1.2 mi
2019/10/18 12:50:48.753604 [NARRATIVE]    VERBAL_ALERT: Turn left onto Green Valley Road.
2019/10/18 12:50:48.753615 [NARRATIVE]    VERBAL_PRE: Turn left onto Green Valley Road.
2019/10/18 12:50:48.753626 [NARRATIVE]    VERBAL_POST: Continue for 1.2 miles.
2019/10/18 12:50:48.753635 [NARRATIVE] ----------------------------------------------
2019/10/18 12:50:48.753648 [NARRATIVE] 4: Turn left onto Glenmill Road. | 0.1 mi
2019/10/18 12:50:48.753683 [NARRATIVE]    VERBAL_ALERT: Turn left onto Glenmill Road.
2019/10/18 12:50:48.753702 [NARRATIVE]    VERBAL_PRE: Turn left onto Glenmill Road.
2019/10/18 12:50:48.753713 [NARRATIVE]    VERBAL_POST: Continue for 1 tenth of a mile.
2019/10/18 12:50:48.753722 [NARRATIVE] ----------------------------------------------
2019/10/18 12:50:48.753735 [NARRATIVE] 5: Turn right onto Spickert Knob Road. | 0.8 mi
2019/10/18 12:50:48.753749 [NARRATIVE]    VERBAL_ALERT: Turn right onto Spickert Knob Road.
2019/10/18 12:50:48.753783 [NARRATIVE]    VERBAL_PRE: Turn right onto Spickert Knob Road.
2019/10/18 12:50:48.753796 [NARRATIVE]    VERBAL_POST: Continue for 8 tenths of a mile.
2019/10/18 12:50:48.753807 [NARRATIVE] ----------------------------------------------
2019/10/18 12:50:48.753822 [NARRATIVE] 6: You have arrived at your destination. | 0.0 mi
2019/10/18 12:50:48.753838 [NARRATIVE]    VERBAL_ALERT: You will arrive at your destination.
2019/10/18 12:50:48.753849 [NARRATIVE]    VERBAL_PRE: You have arrived at your destination.
2019/10/18 12:50:48.753858 [NARRATIVE] ==============================================
2019/10/18 12:50:48.753869 [NARRATIVE] Total time: 7 minutes
2019/10/18 12:50:48.753887 [NARRATIVE] Total length: 2.6 mi
2019/10/18 12:50:48.753914 [INFO] trip_processing_time (ms)::56
2019/10/18 12:50:48.753974 [INFO] trip_time (secs)::452
2019/10/18 12:50:48.753987 [INFO] trip_length (meters)::4215.298828
2019/10/18 12:50:48.754034 [INFO] Total time= 197 ms

In the instructions, it is telling me to drive Southeast on State St. Shouldn't it have given me a U-turn instruction instead?

**Comment by kevinkreiser** [comment]
your point is almost directly between two oneway sections of road. if you dont provide a radius to the  location it will try to use the closest road to the input point. in this case the closest one is the southeast direction of the road. the heading filter is applied last. so if the only candidate you have is the southeast section because you havent given a radius, then it will say well, since we only have this one edge lets just ignore the heading because giving back a route is better than none.

long story short, if you want to pass ambiguous points, you should also pass a radius that covers the area of ambiguity. i havent tested it, but if you had a 15 meter radius i would expect it to do what you want it to do.

**Comment by Jkkarr2015** [comment]
@kevinkreiser I added the 15-meter radius and it did change the message to "make a u-turn" instead. Thanks for the help.
